src/RequestExecutor/HttpRequestExecutor.ts

```diff
@@ -292,8 +295,11 @@ export class HttpRequestExecutor implements RequestExecutor {
   private async parseBody(
     res: IncomingMessage,
     options: {
-      maxBodySize: number;
-      requiresTruncating: boolean;
+      maxSizeWhitelistedMimes: number;
+      maxSizeNonWhitelistedMimes: number;
+      type: string;
+      whiteListed: boolean;
+      allowTruncation: boolean;
```

Comment:
basically this should be enough for parseBody function
it will greatly simplify shouldTruncate and truncateBody
```suggestion
      maxSize: number;
      allowTruncation: boolean;
```